x/name/types/msgs.go

```diff
@@ -101,3 +103,65 @@ func (msg MsgDeleteNameRequest) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{addr}
 }
+
+// NewMsgCreateRootNameRequest creates a new Create Root Name Request
+func NewMsgCreateRootNameRequest(title string, description string, name string, owner string, restricted bool, fromAddress string) *MsgCreateRootNameRequest {
+	return &MsgCreateRootNameRequest{
+		Title:       title,
+		Description: description,
+		Name:        name,
+		Owner:       owner,
+		Restricted:  restricted,
+		FromAddress: fromAddress,
+	}
+}
+
+// Route implements Msg
+func (msg MsgCreateRootNameRequest) Route() string { return ModuleName }
```

Comment:
The only funcs you need to implement for the `Msg` interface are:
```golang
ValidateBasic() error
GetSigners() []AccAddress
```
(Also `String() string` if that's not created automatically from the proto).

But you no longer need `Route`, `Type`, or `GetSignBytes`.

Since we don't need to implement v1beta1 `Content` either, you also don't need:
```golang
GetTitle() string
GetDescription() string
ProposalRoute() string
ProposalType() string
```